### PR TITLE
 CLN: Added static types _libs/algos

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/algos.pyx [hunk 1]
```diff
@@ -261,7 +260,7 @@ def kth_smallest(numeric[:] a, Py_ssize_t k) -> numeric:
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def nancorr(const float64_t[:, :] mat, bint cov=0, minp=None):
+def nancorr(const float64_t[:, :] mat, bint cov=False, minp=None):
```
Comment (MomIsBestFriend) [inline pandas/_libs/algos.pyx:263]: The generated C code for this was not changed

File: pandas/_libs/algos.pyx [hunk 2]
```diff
@@ -810,18 +812,14 @@ def rank_1d(
...
         ndarray[float64_t] ranks
         ndarray[int64_t] argsorted
         ndarray[uint8_t, cast=True] sorted_mask
-
         rank_t val, nan_value
-
         float64_t sum_ranks = 0
         int tiebreak = 0
-        bint keep_na = 0
+        bint keep_na = False
```
Comment (MomIsBestFriend) [inline pandas/_libs/algos.pyx:822]: The generated C code for this was not changed

File: pandas/_libs/algos.pyx [hunk 3]
```diff
@@ -1034,19 +1032,14 @@ def rank_2d(
...
         ndarray[rank_t, ndim=2] values
-
         ndarray[int64_t, ndim=2] argsorted
-
         rank_t val, nan_value
-
         float64_t sum_ranks = 0
         int tiebreak = 0
-        bint keep_na = 0
+        bint keep_na = False
```
Comment (MomIsBestFriend) [inline pandas/_libs/algos.pyx:1042]: The generated C code for this was not changed

File: pandas/_libs/algos.pyx [hunk 4]
```diff
@@ -325,7 +324,7 @@ def nancorr(const float64_t[:, :] mat, bint cov=0, minp=None):
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def nancorr_spearman(const float64_t[:, :] mat, Py_ssize_t minp=1):
+def nancorr_spearman(const float64_t[:, :] mat, Py_ssize_t minp=1) -> ndarray:
```
Comment (WillAyd) [inline pandas/_libs/algos.pyx:327]: Does adding these return annotations to the def like this change anything?

Comment (jbrockmendel) [reply pandas/_libs/algos.pyx:327]: for functions that arent cdef/cpdef its ornamental

**Review by WillAyd — APPROVED** [review]
Lgtm

**Comment by jreback** [comment]
thanks @MomIsBestFriend

**Pull request merged**